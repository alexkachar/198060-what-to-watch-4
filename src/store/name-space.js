const NameSpace = {
  DATA: `DATA`,
  UI: `UI`,
  USER: `USER`,
  REVIEW: `REVIEW`
};

export default NameSpace;
