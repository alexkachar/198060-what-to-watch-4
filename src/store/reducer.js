import {combineReducers} from "redux";
import data from "./reducers/data/data";
import ui from "./reducers/ui/ui";
import user from "./reducers/user/user";
import NameSpace from "./name-space";

export default combineReducers({
  [NameSpace.DATA]: data,
  [NameSpace.UI]: ui,
  [NameSpace.USER]: user,
});
