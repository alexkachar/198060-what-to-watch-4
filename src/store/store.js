import thunk from 'redux-thunk';
import {compose} from 'recompose';
import {createStore, applyMiddleware} from 'redux';

import createAPI from '../api';
import reducer from './reducer';
import DataOperation from './operations/data/data';
import UserOperation from './operations/user/user';

import UserActionCreator from './actions/user/user';
import {AuthStatus} from '../constants';

const onUnauthorized = () => {
  store.dispatch(UserActionCreator.setAuthStatus(AuthStatus.NO_AUTH));
};

const api = createAPI(onUnauthorized);

const store = createStore(
    reducer,
    compose(
        applyMiddleware(thunk.withExtraArgument(api)),
        window.__REDUX_DEVTOOLS_EXTENSION__ ? window.__REDUX_DEVTOOLS_EXTENSION__() : (f) => f
    )
);

export const initApp = () => {
  store.dispatch(DataOperation.loadMovies());
  store.dispatch(UserOperation.checkAuthStatus());
};

export default store;
