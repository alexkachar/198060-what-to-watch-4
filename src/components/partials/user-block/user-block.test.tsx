import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import store from '../../../test-data/mock-store';

import UserBlock from './user-block';

it(`UserBlock renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <UserBlock />
          </BrowserRouter>
        </Provider>, {
          createNodeMock: () => document.createElement(`div`)
        })
    .toJSON();
  expect(tree).toMatchSnapshot();
});
