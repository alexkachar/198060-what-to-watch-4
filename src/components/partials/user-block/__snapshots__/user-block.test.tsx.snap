// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock renders correctly 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__avatar"
    href="/favorites"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <img
      src="https://htmlacademy-react-3.appspot.com/img/john_doe.jpg"
      style={
        Object {
          "borderRadius": "50%",
          "height": "63px",
          "width": "63px",
        }
      }
    />
  </a>
</div>
`;
